File: 0168-excel-sheet-column-title/0168-excel-sheet-column-title.js
/**
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function(columnNumber) {
  let ans = '';
  let code;
  while (columnNumber > 0) {
    if (columnNumber % 26) {
      code = columnNumber % 26;
      columnNumber = Math.floor(columnNumber / 26);
    } else {
      code = 26;
      columnNumber -= code;
      columnNumber /= code;
    }
    ans = String.fromCharCode('A'.charCodeAt(0) + code - 1) + ans;
  }
  return ans;
};

File: 0168-excel-sheet-column-title/0168-excel-sheet-column-title.java
class Solution {
  public String convertToTitle(int columnNumber) {
    String ans = "";
    int code;
    while (columnNumber > 0) {
      if (columnNumber % 26 > 0) {
        code = columnNumber % 26;
        columnNumber = columnNumber / 26;
      } else {
        code = 26;
        columnNumber -= code;
        columnNumber /= code;
      }
      ans = new Character((char) ((int) 'A' + code - 1)).toString() + ans;
    }
    return ans;
  }
}